### DISCUSSION TITLE
completed branch instead of pull/x/merge

### DISCUSSION THREAD

**Discussion by IciaCarroBarallobre** [actions/checkout#622] [Q&A]
Hi!
 
I'm starting to use GitHub Action and when an action trigger involves a PR, checkout@v2  selected branch are pull/x/merge.

Do you know how to use checkout@v2 with the current branch? (And not /pull/x/merge one)

Thanks in advance!

**Comment by Ghostbird** [comment]
I'm having the same issue with `checkout@v3`. It's using `pull/x/merge`. I don't know how that ref is constructed, but it can't be a normal merge commit. If I locally merge this branch, the end result is correct, but that ref in Github Actions contains lots of build-files that were deleted in the branch.

EDIT: Turned out to be a caching issue. Once I managed to get a completely clean runner, the problem was gone.
